agi-bin/directory.lib.php

```diff
@@ -187,6 +187,21 @@ function readContact($con, $keys='#'){
 					}
 				}
 				//fallthough if not successfull
+			case 'tts':
+				// speak the name if possible, otherwise move on to spell it
+				$tmp = $this->agi_get_var('ASTSPOOLDIR') . '/tmp/directory-tts-' . time() . rand(100, 999);
+				system('flite -t "' . $con['name'] .'" -o ' . $tmp . '.wav', $exit_code);
+				if ($exit_code === 0) {
+						$ret = $this->agi->stream_file($tmp, $keys);
+						if ($ret['result']) {
+							$ret['result'] = chr($ret['result']);
+						}
+						unlink($tmp . '.wav');
```

Comment:
Check to make sure the file was actually generated before unlinking. Might actually want to check on line 194 before streaming the file